service/frontend/fx.go

```diff
@@ -362,27 +362,39 @@ func NamespaceRateLimitInterceptorProvider(
 		panic("invalid service name")
 	}
 
-	rateFn := quotas.ClusterAwareNamespaceSpecificQuotaCalculator{
+	namespaceRateFn := quotas.ClusterAwareNamespaceSpecificQuotaCalculator{
 		MemberCounter:    frontendServiceResolver,
 		PerInstanceQuota: serviceConfig.MaxNamespaceRPSPerInstance,
 		GlobalQuota:      globalNamespaceRPS,
 	}.GetQuota
+	namespaceBurstFn := func(ns string) int {
+		return int(namespaceRateFn(ns) *
+			serviceConfig.MaxNamespaceBurstRatioPerInstance(ns))
+	}
 	visibilityRateFn := quotas.ClusterAwareNamespaceSpecificQuotaCalculator{
 		MemberCounter:    frontendServiceResolver,
 		PerInstanceQuota: serviceConfig.MaxNamespaceVisibilityRPSPerInstance,
 		GlobalQuota:      globalNamespaceVisibilityRPS,
 	}.GetQuota
+	visibilityBurstFn := func(ns string) int {
+		return int(visibilityRateFn(ns) *
+			serviceConfig.MaxNamespaceVisibilityBurstRatioPerInstance(ns))
+	}
 	namespaceReplicationInducingRateFn := quotas.ClusterAwareNamespaceSpecificQuotaCalculator{
 		MemberCounter:    frontendServiceResolver,
 		PerInstanceQuota: serviceConfig.MaxNamespaceNamespaceReplicationInducingAPIsRPSPerInstance,
 		GlobalQuota:      globalNamespaceNamespaceReplicationInducingAPIsRPS,
 	}.GetQuota
+	namespaceReplicationInducingBurstFn := func(ns string) int {
+		return int(namespaceReplicationInducingRateFn(ns) *
+			serviceConfig.MaxNamespaceNamespaceReplicationInducingAPIsBurstRatioPerInstance(ns))
+	}
 	namespaceRateLimiter := quotas.NewNamespaceRequestRateLimiter(
 		func(req quotas.Request) quotas.RequestRateLimiter {
 			return configs.NewRequestToRateLimiter(
-				configs.NewNamespaceRateBurst(req.Caller, rateFn, serviceConfig.MaxNamespaceBurstPerInstance),
-				configs.NewNamespaceRateBurst(req.Caller, visibilityRateFn, serviceConfig.MaxNamespaceVisibilityBurstPerInstance),
-				configs.NewNamespaceRateBurst(req.Caller, namespaceReplicationInducingRateFn, serviceConfig.MaxNamespaceNamespaceReplicationInducingAPIsBurstPerInstance),
+				configs.NewNamespaceRateBurst(req.Caller, namespaceRateFn, namespaceBurstFn),
```

Comment:
nit: looks like we can change NewNamespaceRateBurst to take in ratio instead of a burst fn to save some duplicated logic for defining the burstFn. 

It may be worth to do some validation on the ratio to make sure things like burst (token bucket size) is at least 1. 
https://github.com/temporalio/temporal/blob/a2eb30181e85baa88a41a12e205d12a480f1730a/common/quotas/dynamic.go#L90

Reply:
what about we always take `BurstRatio = min(1, BurstRatio)`. This way burst is atleast RPS. Reducing burst below 1 will reduce rps. If someone wants to reduce RPS they should not use burst config for that. This will prevent burst from accidently reducing RPS.